ocs_ci/ocs/cluster.py

```diff
@@ -912,6 +913,85 @@ def get_osd_utilization():
+    output = ct_pod.exec_ceph_cmd(ceph_cmd=ceph_cmd)
+    return output
+
+
+def validate_replica_data(pool_name, replica):
+    """
+    Check if data is replica 2 or 3
+    :param replica: size of the replica(2,3)
+    :type replica: int
+    :param pool_name: name of the pool to check replica
+    :type pool_name: str
+    :return: true if replica is verified, false if not
+    :rtype: bool
+    """
+    from ocs_ci.ocs.exceptions import PoolNotFound
+    output = get_ceph_df_detail()
+    pool_list = output.get('pools')
+    for pool in pool_list:
+        if pool.get('name') == pool_name:
+            logger.info(f"{pool_name}")
+            stored = pool.get('stats').get('stored')
+            byte_used = pool.get('stats').get('bytes_used')
+            compress_bytes_used = pool.get('stats').\
+                get('compress_bytes_used')
+            compress_under_bytes = pool.get('stats').\
+                get('compress_under_bytes')
+            byte_used = byte_used + compress_under_bytes - compress_bytes_used
+            store_ratio = byte_used / stored
+            if (replica + 0.2) > store_ratio > (replica - 0.2):
+                logger.info(f"pool {pool_name} meet rep {replica} size")
+                return True
+            else:
+                logger.info(f"pool {pool_name} meet do not meet rep"
+                            f" {replica} size"
+                            f"Store ratio is {store_ratio}")
+                return False
+        if pool == pool_list[-1]:
+            raise PoolNotFound(f"Pool {pool_name} not found on cluster")
+
+
+def validate_compression(pool_name):
+    """
+    Check if data was compressed
+    :param pool_name: name of the pool to check replica
+    :type pool_name: str
+    :return: True if compression ratio is right, false if compression
+    ratio is wrong
+    :rtype: bool
+    """
+    output = get_ceph_df_detail()
+    pool_list = output.get('pools')
+    for pool in pool_list:
+        if pool.get('name') == pool_name:
+            logger.info(f"{pool_name}")
+            byte_used = pool.get('stats').get('bytes_used')
+            compress_bytes_used = pool.get('stats').get('compress_bytes_used')
+            compress_under_bytes = pool.get('stats').get('compress_under_bytes')
+            all_byte_used = byte_used + compress_under_bytes - compress_bytes_used
+            compression_ratio = byte_used / all_byte_used
+            logger.info(f"this is the comp_ratio {compression_ratio}")
+            if 0.6 < compression_ratio:
+                logger.info(f"Compression ratio {compression_ratio} is "
+                            f"larger than 0.6")
+                return True
+            else:
+                logger.info(f"Compression ratio {compression_ratio} is "
+                            f"smaller than 0.6")
+                return False
+        if pool == pool_list[-1]:
+            raise PoolNotFound(f"Pool {pool_name} not found on cluster")
```

Comment:
As I noted in another thread, let's just omit the if and raise the exception directly. The code there won't be executed if the pool is found.

Reply:
remove the if and raise outside loop